File: core/src/com/kurante/projectvoice_gdx/util/ScaleLabelByAction.java
package com.kurante.projectvoice_gdx.util;

import com.badlogic.gdx.scenes.scene2d.actions.RelativeTemporalAction;
import com.badlogic.gdx.scenes.scene2d.ui.Label;

public class ScaleLabelByAction extends RelativeTemporalAction {
    private float amountX, amountY;

    protected void updateRelative (float percentDelta) {
        ((Label)target).setFontScale(amountX * percentDelta, amountY * percentDelta);
    }

    public void setAmount (float x, float y) {
        amountX = x;
        amountY = y;
    }

    public void setAmount (float scale) {
        amountX = scale;
        amountY = scale;
    }

    public float getAmountX () {
        return amountX;
    }

    public void setAmountX (float x) {
        this.amountX = x;
    }

    public float getAmountY () {
        return amountY;
    }

    public void setAmountY (float y) {
        this.amountY = y;
    }

}

File: core/src/com/kurante/projectvoice_gdx/util/ScaleLabelByAction.kt
package com.kurante.projectvoice_gdx.util

import com.badlogic.gdx.scenes.scene2d.actions.RelativeTemporalAction
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.kurante.projectvoice_gdx.util.extensions.scaleFontBy

class ScaleLabelByAction : RelativeTemporalAction() {
    var amountX = 0f
    var amountY = 0f
    override fun updateRelative(percentDelta: Float) {
        (target as Label).scaleFontBy(amountX * percentDelta, amountY * percentDelta)
    }

    fun setAmount(x: Float, y: Float) {
        amountX = x
        amountY = y
    }

    fun setAmount(scale: Float) {
        amountX = scale
        amountY = scale
    }
}